### PR TITLE
Update shell completion command

### PR DESCRIPTION
**Pull request by eljhkrr**
When we moved to minimist shell completion for medic-conf commands broke. This fixes completion and accommodates users who may not have an updated bash profile.

### CODE REVIEW

**Comment by eljhkrr** [comment]
We only default to bash if the user provides an unrecognized shell type. Ideally, we should automatically detect the user's shell and default to bash if we don't find a compatible type. However, there are simply too many variations to keep up with and since most users tend to use mainstream flavors, I think we're safe in using bash as the baseline for now.

**Comment by eljhkrr** [comment]
Look at me making assumptions about users. We now throw an error if an unsupported shell is encountered. @kennsippell please review.

**Comment by eljhkrr** [comment]
Apologies for sporadic updates on this since requesting review. This is now complete and ready for review.

**Comment by kennsippell** [comment]
This one fell through the cracks. Will look tomorrow @eljhkrr!

File: test/bin/shell-completion.spec.js [hunk 1]
```diff
@@ -1,70 +1,22 @@
...
-  it('should supply --options and actions', () => {
+  it('should complete a partial command', () => {
 
-    // expect
-    return assertShellComplete('',
-        '--instance',
-        '--local',
-        '--url',
-        '--user',
+    let partial = 'comp';
```
Comment (kennsippell) [inline test/bin/shell-completion.spec.js]: These seem like `const`

File: test/bin/shell-completion.spec.js [hunk 2]
```diff
@@ -1,70 +1,22 @@
...
+    let bin = 'medic-conf';
+    let input = bin + ' ' + partial;
 
-        '--help',
-        '--shell-completion',
-        '--supported-actions',
-        '--version',
-        '--accept-self-signed-certs',
+    let cmd = `bash -i -c 'source /etc/bash_completion; COMP_LINE="${input}"; COMP_WORDS=(${input}); COMP_CWORD=1; COMP_POINT=${input.length}; $(complete -p medic-conf | sed "s/.*-F \\([^ ]*\\) .*/\\1/") && echo \$\{COMPREPLY[*]\}'`; // eslint-disable-line no-useless-escape
```
Comment (kennsippell) [inline test/bin/shell-completion.spec.js]: Consider making this a multi-line string.
I don't really understand what it does, so maybe some comments.

File: README.md [hunk 3]
```diff
@@ -81,6 +81,77 @@ The list of available actions can be seen via `medic-conf --help`.
 
 	medic-conf <--local|--instance=instance-name|--url=url> <...action> -- <...form>
 
+# Project Layout
```
[the remark this answers is not in the document]
Comment (eljhkrr) [reply README.md]: Yeah, bad merge. Fixed.

File: test/bin/shell-completion.spec.js (hunk 2 above)
Comment (eljhkrr) [reply test/bin/shell-completion.spec.js]: Updated

File: src/lib/main.js [hunk 4]
```diff
@@ -53,7 +53,12 @@ module.exports = async (argv, env) => {
   }
 
   if (cmdArgs['shell-completion']) {
-    return shellCompletionSetup(cmdArgs['shell-completion']);
+    if (cmdArgs['shell-completion'] === true){
+      console.log('# ERROR shell type argument not provided e.g. --shell-completion=bash');
```
[the remark this answers is not in the document]
Comment (eljhkrr) [reply src/lib/main.js]: Yes, this case is handled in `shellCompletionSetup`, although the resulting error message wasn't useful
```
# ERROR medic-conf shell completion not yet supported for true
```
I've moved the logic for differentiating error messages for the two scenarios: leaving out the shell argument vs specifying an unsupported shell to the `shellCompletionSetup` module as it fits better there.

[the remark this answers is not in the document]
Comment (kennsippell) [reply src/lib/main.js]: Can you encapsulate it in the shell completion file? Trying to keep `main` as simple as possible, and encapsulate action-specific logic into their respective files.

[the remark this answers is not in the document]
Comment (eljhkrr) [reply src/lib/main.js]: Done!

**Comment by garethbowen** [comment]
@kennsippell This was approved but never merged. Can you look into resolving the conflicts and getting it in?

**Comment by garethbowen** [comment]
@kennsippell Is this ready to merge now?

**Comment by kennsippell** [comment]
@garethbowen It was failing the audit check last time I looked, but it looks like @eljhkrr fixed that! Merging!

**Pull request merged**